fidouaf/src/main/java/org/ebayopensource/fidouaf/res/FidoUafResource.java

```diff
@@ -146,8 +172,22 @@ public Facets facets() {
 		return facets;
 	}
 
-	private String getAppId() {
-		return "https://www.head2toes.org/fidouaf/v1/public/uaf/facets";
+	/**
+	 * The AppID is an identifier for a set of different Facets of a relying
+	 * party's application. The AppID is a URL pointing to the TrustedFacets,
+	 * i.e. list of FacetIDs related to this AppID.
+	 * @return a URL pointing to the TrustedFacets
+	 */
+	private String getAppId(HttpServletRequest httpServletRequest) {

```

Comment:
If UriInfo is injected, this method could be:

	private String getAppId() {
		return uriInfo.getBaseUri() + "v1/public/uaf/facets";
//		return "https://www.head2toes.org/fidouaf/v1/public/uaf/facets";
	}

Also, there would be no need to pass the UriInfo as a parameter.